src/pick.ts

```diff
@@ -10,7 +10,7 @@ import { purry } from './purry';
  * @data_first
  * @category Object
  */
-export function pick<T extends {}, K extends keyof T>(
+export function pick<T extends object, K extends keyof T>(
```

Comment:
why would we do `extends object` here instead of `extends Record<PropertyKey, unknown>` ?

Reply:
Because according to the `pick` tests, a class object instance can be passed as a param and not only objects in `{ property: value }` format.
[This is the snippet that explains what I want to say](https://github.com/remeda/remeda/blob/78a2b12bfee321c334205f352e7a7c15045f3aa3/src/pick.test.ts#L14-L22)

Reply:
And `extends object` is better because it avoids wrongs types in param.
[You can see an example here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAlpKBDMwCmMC8MxoO4wCqASgDIAUA5ABZRQAOAXAPTMDmIIbANmgHSgAtpQCUAbgCwAKGkATNMG5IAThgBmAV1RQ44GPTjAA1gB4AKjDQAPKGjCyIMAN4BfADQwA0pZt2HMIzQATxA1GDMAPnIQACMAKwUoRnC3MCRBNAhk1SRZcG4grwBtAF0RZIAFQ1MzD08IyRkpVnC0EGVDJG4CmGgOsDYYYBQwEFgYjHokCAg0WRhqNFVpA2MqJA2kSg8iyhRganbKMpXq8gRofbQdmhBoNIzjkVO1pxx0tGTKAClwShh3DBdg80E9pHIFEpVDBNNpdGAYAB9VY1Hy2eyOWIJYBQOpovyOQIhMKRaLxRLJWogrIwHJ5MA9TylcowKrGcx1Brg5rMGAAdTgUGo+IxMCxiTFDMK4px8EQV0cwwRE3001msmkyLOe022yBe1Qh2UYKkWrWF2QqGu+sO9w+JrNRnIbxBX1+YH+gOB9rKQA)

Reply:
ooh, my fork wasn't up to date, so I didn't have that test, which made me curious 😅 

👍